sphinx_gallery/utils.py

```diff
@@ -90,3 +91,27 @@ def replace_py_ipynb(fname):
             % (allowed_extension, extension))
     new_extension = '.ipynb'
     return '{}{}'.format(fname_prefix, new_extension)
+
+
+def get_md5sum(src_file):
+    """Returns md5sum of file"""
+    with open(src_file, 'rb') as src_data:
+        src_content = src_data.read()
+        return hashlib.md5(src_content).hexdigest()
+
+
+def _replace_md5(fname_new, fname_old=None, method='move'):
+    assert method in ('move', 'copy')
+    if fname_old is None:
+        assert fname_new.endswith('.new')
+        fname_old = fname_new[:-4]
```

Comment:
small nitpick, in my experience this is a bit more dangerous than just doing `os.path.splitext(fname_new)[0]`...it seems trivial but I've been bitten by it before when I change code in the future